### DISCUSSION TITLE
overlapping key mappings

### DISCUSSION THREAD

**Discussion by BommerG** [charmbracelet/bubbletea#900] [Q&A]
Hi,

I'm trying to develop an application that has multiple tabs and each tab can have multiple input elements (textinput, list, ...). so, I need keys to move between the tabs and the different input elements. Now my problem is that e.g. the left/right keys to move between the tabs are also used by the textinput bubble and the keys up/down to move between the different input elements are also used by the list bubble.

any idea how to solve this problem in a consistent way?

I was thinking about delivering the KeyMsg to the currently active input element. the element is then  free to consume the message or to ignore it. in the later case the message would be routed to the parent. but there is no easy way to know if the input element reacted to a certain message or just ignored it.

On the other hand, if the parent consumes its keys first then the functionality of the lower elements could be impaired because they never see certain messages.

of course I could create a key map without overlapping keys but this will result in an application that is not intuitive to use.

Any input is greatly appreciated
Cheers
Boomer

**Comment by KevM** [comment]
A few thoughts: 
1. Avoid this as it is likely confusing for your users. 
2. You could have a first in wins for Updates handling`tea.KeyMsg` they could return their command and not let the other updates funcs receive the msg thus controlling which component "wins" the keypress.
